ctapipe/image/timing.py

```diff
@@ -4,17 +4,91 @@
 
 import numpy as np
 import astropy.units as u
-from numpy.polynomial.polynomial import polyval
+from numba import njit
+
 from ..containers import TimingParametersContainer
 from .hillas import camera_to_shower_coordinates
 from ..utils.quantities import all_to_value
 
-from scipy.stats import siegelslopes
-
 
 __all__ = ["timing_parameters"]
 
 
+@njit(cache=True)
+def linear_regression(x, y, weights):
+    """
+    njit version of a least squares linear regression
+
+    Parameters
+    ----------
+
+    x: np.ndarray
+        x values
+    y: np.ndarray
+        y values
+
+    Returns
+    -------
+    slope: float
+        slope of the linear regression result
+    intercept: float
+        intercept of the linear regression result
+    cov: 2x2 ndarray
+        covariance matrix of the slope and intercept
+    """
+    y = y.reshape((-1, 1))
+    X = np.empty((len(x), 2))
+    X[:, 0] = x
+    X[:, 1] = 1
+    W = np.diag(weights)
+
+    cov = np.linalg.inv(X.T @ W @ X)
```

Comment:
I suggest using `np.linalg.pinv()` here instead (pseudo-inverse, using SVD) That will never fail,  and is more stable.